internal/engine/upper.go

```diff
@@ -133,28 +133,36 @@ func loadAndGetManifests(ctx context.Context, manifestNames []model.ManifestName
 
 	tf, err := tiltfile.Load(ctx, tiltfile.FileName)
 	if os.IsNotExist(err) {
-		manifests = []model.Manifest{}
 		globalYAML = model.YAMLManifest{}
 	} else if err != nil {
 		return nil, model.YAMLManifest{}, nil, err
 	} else {
 		manifests, globalYAML, configFiles, err = tf.GetManifestConfigsAndGlobalYAML(ctx, manifestNames...)
 		if err != nil {
-			manifests = []model.Manifest{}
 			globalYAML = model.YAMLManifest{}
 		}
 	}
 
+	manifests = append(
+		manifests,
+		model.Manifest{
```

Comment:
i'm very skeptical of this data model. i think it's going to lead to way too many bugs down the road. @landism, what do you think?

Reply:
Skeptical about what in particular? That we append the TiltfileManifest to the list of regular manifests?

Reply:
yes, and just the idea that we model Tiltfile execution this way

Reply:
Maybe the four of us (you, me, @landism and @dbentley) should get together after standup to talk trough the plan. My understanding from @dbentley is that there will be more such "special manifests in the future.